flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/GSAGreedyGraphColoringExample.java

```diff
@@ -0,0 +1,212 @@
+		@Override
+		public HashSet<Double> gather(Triplet<Double, Double> triplet) {
+
+			HashSet<Double> result = new HashSet<Double>();
+			result.add(triplet.getSrcVertexValue());
+
+			return result;
+		}
+	};
+
+	private static final class GreedyGraphColoringSum
+			extends SumFunction<Double, Double, HashSet<Double>> {
+		@Override
+		public HashSet<Double> sum(HashSet<Double> newValue, HashSet<Double> currentValue) {
+
+			HashSet<Double> result = new HashSet<Double>();
+			result.addAll(newValue);
+			result.addAll(currentValue);
+
+			return result;
+		}
+	};
+
+	private static final class GreedyGraphColoringApply
+			extends ApplyFunction<Double, Double, HashSet<Double>> {
+		@Override
+		public void apply(HashSet<Double> set, Double src) {
+			double minValue = src;
+			for (Double d : set) {
+				if (d < minValue) {
+					minValue = d;
+				}
+			}
+
+			// This is the condition that enables the termination of the iteration
+			if (minValue < src) {
+				setResult(minValue);
+			}
+		}
+	};
+
+	// --------------------------------------------------------------------------------------------
+	//  Util methods
+	// --------------------------------------------------------------------------------------------
+
+	private static boolean fileOutput = false;
+	private static String vertexInputPath = null;
+	private static String edgeInputPath = null;
+	private static String outputPath = null;
+
+	private static int maxIterations = 16;
+
+	private static boolean parseParameters(String[] args) {
+
+		if(args.length > 0) {
+			// parse input arguments
+			fileOutput = true;
+
+			if(args.length != 4) {
+				System.err.println("Usage: GSAGreedyGraphColoringExample <vertex path> <edge path> " +
+						"<result path> <max iterations>");
+				return false;
+			}
+
+			vertexInputPath = args[0];
+			edgeInputPath = args[1];
+			outputPath = args[2];
+			maxIterations = Integer.parseInt(args[3]);
+		} else {
+			System.out.println("Executing GSA Greedy Graph Coloring example with built-in default data.");
+			System.out.println("  Provide parameters to read input data from files.");
+			System.out.println("  See the documentation for the correct format of input files.");
+			System.out.println("  Usage: GSAGreedyGraphColoringExample <vertex path> <edge path> "
+					+ "<result path> <max iterations>");
+		}
+		return true;
+	}
+
+	private static DataSet<Vertex<Long, Double>> getVertexDataSet(ExecutionEnvironment env) {
+		if(fileOutput) {
+			return env
+					.readCsvFile(vertexInputPath)
+					.fieldDelimiter(" ")
+					.lineDelimiter("\n")
+					.types(Long.class, Double.class)
+					.map(new MapFunction<Tuple2<Long, Double>, Vertex<Long, Double>>() {
+						@Override
+						public Vertex<Long, Double> map(Tuple2<Long, Double> value) throws Exception {
+							return new Vertex<Long, Double>(value.f0, value.f1);
+						}
+					});
+		} else {
+			return ExampleUtils.getLongDoubleVertexData(env);
```

Comment:
This is just a cosmetic bug, but I think it could be useful to mention it... 
Could you, instead of ExampleUtils, take the data from a separate class? For SSSP for example, you have SingleSourceShortestPathsData in example/utils. This would be more legible(i.e. it would look nicer than getLongLongVertexData :) ).
Also @vasia , I see nobody else is using that method in ExampleUtils so maybe after this we can get rid of it :D